scripts/gpu_support/nvidia/link_nvidia_host_libraries.sh

```diff
@@ -484,12 +517,30 @@ symlink_mode () {
         # Loop over each matched library
         for library in "${MATCHED_LIBRARIES[@]}"; do
             log_verbose "Linking library: ${library}"
+            
+            # Get just the library filename
+            lib_name=$(basename "$library")
+            
+            # Check if the symlink already exists
+            if [ -L "$lib_name" ]; then
+                # Check if it's pointing to the same target
+                target=$(readlink "$lib_name")
+                if [ "$target" = "$library" ]; then
+                    log_verbose "Symlink for $lib_name already exists and points to correct target"
+                    continue
+                else
+                    log_verbose "Symlink for $lib_name exists but points to wrong target: $target, updating..."
+                    rm "$lib_name"
+                fi
+            fi
     
             # Create a symlink in the current directory
             # and check if the symlink was created successfully
             if ! ln -s "$library" .
             then
-                fatal_error "Error: Failed to create symlink for library $library in $PWD"
+                echo_yellow "Warning: Failed to create symlink for library $library in $PWD"
+                # Continue instead of fatal_error to make the script more robust
+                continue
```

Comment:
And what happens if the creation of all symlinks fails? I don't think this helps with robustness, to me this is an error.

Reply:
Reverted it back to fatal_error.